### PR TITLE
#10247 S2S Surveillance Reports should be shareable (along with possibly attached External Messages)

### LINKED ISSUE #10247: S2S Surveillance Reports should be shareable (along with possibly attached External Messages) [8]
### Feature Description

prerequisite: #6430 
Currently it is not possible to share physician's reports/ surveillance reports. If you share a case with S2S with hand over the ownership there is no possibility for this. Also the entity 'reports' in general can not be shared. This feature should allow that.

### Proposed Change

- [x] Introduce new properties
1. feature type: SORMAS_TO_SORMAS_SHARE_CASES 
property: SHARE_REPORTS

- [x] change UI of share window 
add a checkbox _share reports_ (see picture below)
- [x] change behaviour of checkboxes (summary: same behaviour as for checkbox samples)

| checkbox | value | behaviour checkbox _share reports_ |
| --- | --- | --- |
| Hand over the ownership | true | is checked and greyed out, _reports_ box surveillance reports AND attached physician's report / lab messages have to be shared with | 
| Hand over the ownership | false | is not checked and not greyed out. So checkbox can be checked BUT if checkbox is checked ONLY the box _reports_ is shared NOT the physician's report/ lab message | 
| Exclude personal data | true | is not checked and not greyed out. So checkbox can be checked BUT if checkbox is checked ONLY the box _reports_ is shared NOT the physician's report/ surveillance reports <br/> PLUS please check which fields have to be excluded (from the _reports_ box). I think only SurveillanceReport.facilityDetails and SurveillanceReport.notificationDetails but i do not have a current manual so please check this. Thank you | 
| Exclude personal data | false | no effect | 
| Share samples | true, false | no effect |
| Share immunization | true, false | no effect |
| Share reports | true, false | by clicking, checkbox gets checked or unchecked depending on the previous status. Exception: checkbox is greyed out because of behaviour mentioned above. If checkbox is greyed out, the checkbox is not clickable. |

- [x] If a share was initiated _with hand over the ownership_ and _share reports: true_ the source system is still able to open the 'edit report' window and is able to read what is written. Furthermore they can open the attached physician's reports/ lab messages
- [x] If a share was initiated hand over the ownership= false and share reports=true, the target system is still able to open the 'edit report' window and is able to read what is written. 
- [x] If a share was initiated hand over the ownership=false and share reports=false, the target system does not receive reports and the source system is still able to open the 'edit report' window and is able to edit. 
- [x] shared physician reports / lab messages should also be shown in both system's message directories.
### Implementation Details

image

### Additional Information

- [x] Like the entities _samples_ and _vaccination_: As long as a share request has the status _pending_, the source system can edit and change data. The edited and changed data should be synced immediately.
- [x] Sync should work between both systems (similar to how it is done for samples).

### PR DESCRIPTION
**Pull request by leventegal-she**
Fixes #10247

### CODE REVIEW

**Review by JonasCir — CHANGES_REQUESTED** [review]
Nicely done! 
1. One inline comment
2. Please add a test to `InfraValidationSoundnessTest` similar to `testShareLabMessageValidation`
3. PR needs rebase
4. I assume this should only merged for the next release I guess?

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/surveillancereport/SurveillanceReportDto.java [hunk 1]
```diff
@@ -51,23 +52,24 @@ public class SurveillanceReportDto extends PseudonymizableDto {
...
 	public static final String NOTIFICATION_DETAILS = "notificationDetails";
+	private UserReferenceDto reportingUser;
 
-	public static SurveillanceReportDto build(CaseReferenceDto caze, UserReferenceDto creatingUser) {
+	@NotNull
+	private ReportingType reportingType;
+
+	private String externalId;
+
+	public static SurveillanceReportDto build(CaseReferenceDto caze, UserReferenceDto reportingUser) {
```
[the remark this answers is not in the document]
Comment (leventegal-she) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/surveillancereport/SurveillanceReportDto.java:62]: There is no clear default value for those fields, it's better to let the user set theses values.

For example the reporting date on case is required but the re sui no `CaseDataDto::build` function that sets it.

**Comment by leventegal-she** [comment]
> Nicely done!
> 
> 1. One inline comment
> 2. Please add a test to `InfraValidationSoundnessTest` similar to `testShareLabMessageValidation`
> 3. PR needs rebase
> 4. I assume this should only merged for the next release I guess?

1. Responded on the message
2. Done. Can you please add a comment on the test class so me and others would see what those tests do
3. Done
4. Yes, it's for 1.79

File: sormas-api/src/main/java/de/symeda/sormas/api/caze/surveillancereport/SurveillanceReportDto.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (JonasCir) [reply sormas-api/src/main/java/de/symeda/sormas/api/caze/surveillancereport/SurveillanceReportDto.java:62]: ok thanks

**Review by JonasCir — APPROVED** [review]
lgtm, I'm going to add a comment to the test classes later

**Pull request merged**